src/stepfunctions/steps/sagemaker.py

```diff
@@ -284,20 +286,10 @@ def __init__(self, state_id, model, model_name=None, instance_type=None, tags=No
             instance_type (str, optional): The EC2 instance type to deploy this Model to. For example, 'ml.p2.xlarge'.
             tags (list[dict], optional): `List to tags <https://docs.aws.amazon.com/sagemaker/latest/dg/API_Tag.html>`_ to associate with the resource.
         """
-        if isinstance(model, FrameworkModel):
+        if isinstance(model, Model):
             parameters = model_config(model=model, instance_type=instance_type, role=model.role, image_uri=model.image_uri)
             if model_name:
                 parameters['ModelName'] = model_name
-        elif isinstance(model, Model):
```

Comment:
Is this backwards compatible?

Reply:
yes, this generated the same parameters as the old way of doing, but takes into account the env variables as well

Reply:
Not quite. If `model` is the `Model` base class, parameters will have new fields that weren't there before. It's not just `PrimaryContainer.Evnironment`, but `VpcConfig` and other fields in the [container definition](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ContainerDefinition.html).

https://github.com/aws/sagemaker-python-sdk/blob/e78d0ead32ebd163a21996be44c8cd9a5ee31379/src/sagemaker/workflow/airflow.py#L566-L605

I don't have enough context to know what effect that has, but instantiating a ModelStep with the same model will not produce the same parameters as before.

Reply:
You are right - it shouldn't produce the same parameters if `vpc_config` or an `image_config` (see [here](https://github.com/aws/sagemaker-python-sdk/blob/e78d0ead32ebd163a21996be44c8cd9a5ee31379/src/sagemaker/session.py#L4195-L4224)) were provided in the Model as they would be included in the parameters as well.

Reply:
It is a good point to consider why those params were initially omitted, but I think it makes sense to have consistency between parameters generated from FrameworkModel with the ones generated from Model

With a FrameworkModel, params would also include `vpc_config`, but not `image_config` (container_def() is called without `image_config` arg [here](https://github.com/aws/sagemaker-python-sdk/blob/e78d0ead32ebd163a21996be44c8cd9a5ee31379/src/sagemaker/workflow/airflow.py#L563))

Reply:
> it shouldn't produce the same parameters if vpc_config or an image_config (see here) were provided 

can we add the tests that *would* have caught this. our coverage might be limited 

> It is a good point to consider why those params were initially omitted,

Do the commits where they were introduced provide that context? In any case, we need to preserve existing behaviour. Customers who upgrade without changing any of their code should be able to do that without unexpected mutations.

Reply:
>can we add the tests that would have caught this. our coverage might be limited

Yes, i will add test to ensure there is no regression and/or breaking changes

>Do the commits where they were introduced provide that context? In any case, we need to preserve existing behaviour. Customers who upgrade without changing any of their code should be able to do that without unexpected mutations.

It was included in the initial commit, so no insight on why they were omitted